ddcommon/src/lib.rs

```diff
@@ -62,6 +62,13 @@ struct SerializedUri<'a> {
     path_and_query: Option<Cow<'a, str>>,
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+enum StringOrSerializedUri<'a> {
+    String(String),
+    SerializedUri(SerializedUri<'a>),
+}
```

Comment:
There is a reason why this is serialized like that and not just as a String (which it used to in the past).

The reason is that non http uris (unix sockets unix:// , windows pipes, or file uris) are not correctly deserialized by the hyper::Uri from string implementation.

Could you add unit tests to check what happens when you try to deserialize non http Uris?
I'm interested in particular in paths starting with `/`, `./` and containing spaces as they have caused problems

Reply:
Because this is passed to Hyper directly, I assumed it was just a shortcut to serialize `Url` which is not serializable by default. If there is a good reason to pass a more complex object, I'm fine with also passing that structure instead of relying on a string and removing this change. However, I tend to prefer handling this kind of common use cases internally behind the API instead of externally, thus exposing Hyper limitations to the caller.

Reply:
Actually, how do I even get the tests to pass? Whatever I do it seems there are a lot of tests that are failing before I even touch anything, and I'm unable to make them pass.